layers/+lang/haskell/README.org

```diff
@@ -342,6 +359,47 @@ type is inserted. On a symbol "foo" without definition,
 the case is split. When checking with hlint, original code
 is replaced with hlint's suggestion if possible.
 
+* Syntax checking
+At the moment there are 4 components which can check the syntax and indicates somehow error and warnings in the code.
+Those components are :
+
+
+  - flycheck
+  - hlint (via flycheck)
+  - ghc-mod
+  - haskell-mode interactive
+    
+As all those components can be active at the same time, it can be tricky to know which component is display which message,
+especially when they disagree or one is not working. Only flycheck errors (ghc and hlint) are displayed in the error list and
+can be navigated using the standard spacemacs key bindings even though errors from other mode might highlight error in the buffer.
+
+** Flycheck   
+This is the standard spacemacs way to do syntax checking and the most elaborate. flyckeck displays errors, warning, info, including symbols on the left or right side of the buffer
```

Comment:
I mean, why duplicate info, when we can link to existing Spacemacs documentation (`syntax-checking` layer)? If you think that that layer documentation is missing something - just improve it. My point is that `flycheck` is not `haskell` only thing, it's used all over the places in Spacemacs. 

P. S. don't get me wrong - I am all for better description of Flycheck (in my opinion, it should have a section in general documentation file). Describing haskell-related parts of flycheck should be in haskell layer, obviously, just don't get to general here :)